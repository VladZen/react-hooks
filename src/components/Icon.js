import PropTypes from 'prop-types'

import sprite from '@images/sprite.svg'

const Icon = ({ alias, ...props }) => {
  return (
    <svg {...props}>
      <use href={ `${sprite}#${alias}` }></use>
    </svg>
  )
}

Icon.defaultProps = {
  alias: ''
}

Icon.propTypes = {
  alias: PropTypes.string // could be improved to support a list of items declared in sprite, but it could be hardly maintainable with the growth of the project
}

export default Icon
