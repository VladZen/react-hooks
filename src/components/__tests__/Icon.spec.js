import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import Icon from '@/components/Icon'

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('Icon component', () => {
  describe('rendering', () => {
    it('without children', () => {
      act(() => {
        render(<Icon alias='arrow' />, container)
      })

      expect(container).toMatchSnapshot()
    })

    it('don\'t render any children', () => {
      act(() => {
        render(<Icon alias='arrow'>
          <a href='#'>
            Some link
          </a>
        </Icon>, container)
      })

      expect(container).toMatchSnapshot()
    })

    it('applies props to the root svg', () => {
      act(() => {
        render(<Icon
          alias='arrow'
          className='some-class'
          title='Some title'
        />, container)
      })

      expect(container).toMatchSnapshot()
    })
  })
})
