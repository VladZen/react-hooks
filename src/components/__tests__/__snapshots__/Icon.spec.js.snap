// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component rendering applies props to the root svg 1`] = `
<div>
  <svg
    class="some-class"
    title="Some title"
  >
    <use
      href="sprite.svg#arrow"
    />
  </svg>
</div>
`;

exports[`Icon component rendering don't render any children 1`] = `
<div>
  <svg>
    <use
      href="sprite.svg#arrow"
    />
  </svg>
</div>
`;

exports[`Icon component rendering without children 1`] = `
<div>
  <svg>
    <use
      href="sprite.svg#arrow"
    />
  </svg>
</div>
`;
